server/elasticsearch/src/test/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearchTest.java

```diff
@@ -173,6 +179,58 @@ public void testGetSpanFromEmptyDB() throws StoreException, InterruptedException
         Assert.assertNull(spanFromDB);
     }
 
+    @Test
+    public void testGetTraceFragment() throws StoreException {
+        Span rootSpan = new Span();
+        rootSpan.setId("root");
+        rootSpan.setTraceId("root");
+        rootSpan.setAnnotations(serverAnnotations());
+
+        Span descendant = new Span();
+        descendant.setId("descendant");
+        descendant.setParentId("root");
+        descendant.setTraceId("root");
+        descendant.setAnnotations(clientAnnotations());
+
+        Span descendant2 = new Span();
+        descendant2.setId("descendant2");
+        descendant2.setParentId("root");
+        descendant2.setTraceId("root");
+        descendant2.setAnnotations(clientAnnotations());
+
+        Span descendant3 = new Span();
+        descendant3.setId("descendant3");
+        descendant3.setParentId("root");
+        descendant3.setTraceId("root");
+        descendant3.setAnnotations(serverAnnotations());
```

Comment:
I don't think you could ever get a server span directly contained within another server span.